src/GafferDispatch/Dispatcher.cpp

```diff
@@ -670,6 +676,15 @@ void Dispatcher::dispatch( const std::vector<NodePtr> &nodes ) const
 		return;
 	}
 
+	try
+	{
+		dispatchSignal()( this, taskNodes );
+	}
+	catch( const std::exception& e )
+	{
+		IECore::msg( IECore::Msg::Error, "dispatchSignal exception:", e.what() );
```

Comment:
I wasn't sure if this should be here. I copied it from the postDispatchSignal call site, but its not being done for preDispatchSignal.

Reply:
Yeah, that's an interesting question. I would say :

- We're doing the right thing for `preDispatchSignal()`. This signal is all about cancellation really, especially now we have `dispatchSignal()`. If a connected slot errors, we're better off aborting than we are continuing with a potentially flawed dispatch.
- We're almost doing the right thing for `dispatchSignal()` or `postDispatchSignal()`. Here it's too late to stop the train, so suppressing exceptions from the connected slots is necessary. But we should be doing this using a `CatchingSignalCombiner`, so that if one slot errors, the other slots are still run. This prevents one badly behaved slot from ruining things for all the others.

Related to this, I think there's now a very good case for moving job directory creation _after_ `preDispatchSignal()`. We don't want to produce empty directories in the case of a cancelled dispatch, and any work that needs the directory should now be done in `dispatchSignal()`. Can we add a todo to this effect, so we can follow up once Jabuka is using the new signal?

Reply:
Addressed in 8a5a6057ed6887030bced69f54398ec62d4396fb and 6312ebecfde705952a177264b89693e3796d357a

Reply:
Let's also remove the explicit `try/catch` around the calls to `dispatchSignal/postDispatchSignal` , since those are now redundant, and give a misleading impression.